main.py

```diff
@@ -1,56 +1,249 @@
+    # getting audio file
+    audio_file = voice
+    print('ok_voice')
+    # audio_file = await context.bot.get_file(update.message.voice.file_id)
+
+    # Use a temporary file
+    with tempfile.NamedTemporaryFile(suffix='.wav', delete=True) as temp_audio_file:
+        await audio_file.download_to_drive(custom_path=temp_audio_file.name)
+        chat_id = update.effective_chat.id
+        print('ok2')
+
+        # Convert audio file to base64 encoded string
+        with open(temp_audio_file.name, "rb") as file:
+            audio_data = file.read()
+            audio_base64 = base64.b64encode(audio_data).decode('utf-8')
+
+        response, history = bhashini_audio_chat(chat_id, audio_file=audio_base64, lang=lang)
+        print('ok_')
+        await context.bot.send_message(chat_id=update.effective_chat.id, text=response)
+
+# for getting location,we can use this 
+'''async def location(update: Update, context: ContextTypes.DEFAULT_TYPE) -> int:
+    """Stores the location and asks for some info about the user."""
+    user = update.message.from_user
+    user_location = update.message.location
+    logger.info(
+        "Location of %s: %f / %f", user.first_name, user_location.latitude, user_location.longitude
+    )
+    await update.message.reply_text(
+        "Since you're filing the complaint form your location, we're recording it."
+    )'''
+
+# async def progress_bar(context, chat_id, start_time, update_interval=15, max_duration=90):
+#     while True:
+#         await asyncio.sleep(update_interval)
+#         wait_time = time.time() - start_time
+#         if wait_time > max_duration:
+#             break
+#         await context.bot.send_message(caht_id=chat_id, text="Thank you for your patience. We're wokring on it.")
 
 async def respond(update: Update, context: ContextTypes.DEFAULT_TYPE):
+    start_time = time.time()
     text = update.message.text
     chat_id = update.effective_chat.id
-    print(chat_id)
+    wait_message = get_random_wait_messages(
+        not_always=True
+    )
+    if wait_message:
+        await context.bot.send_message(chat_id=chat_id, text=wait_message)
     response, history = chat(chat_id, text)
+    end_time = time.time()
+    print(f"history status is {history.get('status')}")
+    print(f"Time taken: {end_time - start_time}")
     await context.bot.send_message(chat_id=chat_id, text=response)
 
+
 async def respond_audio(update: Update, context: ContextTypes.DEFAULT_TYPE):
     audio_file = await context.bot.get_file(update.message.voice.file_id)
 
     # Use a temporary file
-    with tempfile.NamedTemporaryFile(suffix='.ogg', delete=True) as temp_audio_file:
+    with tempfile.NamedTemporaryFile(suffix='.wav', delete=True) as temp_audio_file:
         await audio_file.download_to_drive(custom_path=temp_audio_file.name)
         chat_id = update.effective_chat.id
-        print(chat_id)
-        response, history = audio_chat(chat_id, audio_file=open(temp_audio_file.name, "rb"))
```

Comment:
Please make sure existing functions are not edited. Add a new handler function for bhashini if needed.

Reply:
yes